src/sonic-bgpcfgd/tests/test_static_route.py

```diff
@@ -0,0 +1,307 @@
+from unittest.mock import MagicMock, patch
+
+from bgpcfgd.directory import Directory
+from bgpcfgd.template import TemplateFabric
+from bgpcfgd.manager_static_route import StaticRouteMgr
+from collections import Counter
+
+def constructor():
+    cfg_mgr = MagicMock()
+
+    common_objs = {
+        'directory': Directory(),
+        'cfg_mgr':   cfg_mgr,
+        'tf':        TemplateFabric(),
+        'constants': {},
+    }
+
+    mgr = StaticRouteMgr(common_objs, "CONFIG_DB", "STATIC_ROUTE")
+    assert len(mgr.static_routes) == 0
+
+    return mgr
+
+def set_del_test(mgr, op, args, expected_ret, expected_cmds):
+    set_del_test.push_list_called = False
+    def push_list(cmds):
+        set_del_test.push_list_called = True
+        assert Counter(cmds) == Counter(expected_cmds) # check if commands are expected (regardless of the order)
+        max_del_idx = -1
+        min_set_idx = len(cmds)
+        for idx in range(len(cmds)):
+            if cmds[idx].startswith('no') and idx > max_del_idx:
+                max_del_idx = idx
+            if not cmds[idx].startswith('no') and idx < min_set_idx:
+                min_set_idx = idx
+        assert max_del_idx < min_set_idx, "DEL command comes after SET command" # DEL commands should be done first
+        return True
+    mgr.cfg_mgr.push_list = push_list
+
+    if op == "SET":
+        ret = mgr.set_handler(*args)
+        assert ret == expected_ret
+    elif op == "DEL":
+        mgr.del_handler(*args)
+    else:
+        assert False, "Wrong operation"
+
+    if expected_cmds:
+        assert set_del_test.push_list_called, "cfg_mgr.push_list wasn't called"
+    else:
+        assert not set_del_test.push_list_called, "cfg_mgr.push_list was called"
+
+def test_set():
+    mgr = constructor()
+    set_del_test(
+        mgr,
+        "SET",
+        ("10.1.0.0/24", {
+            "nexthop": "10.0.0.57",
+        }),
+        True,
+        [
+            "ip route 10.1.0.0/24 10.0.0.57"
+        ]
+    )
+
+def test_set_nhvrf():
+    mgr = constructor()
+    set_del_test(
+        mgr,
+        "SET",
+        ("default|10.1.1.0/24", {
+            "nexthop": "10.0.0.57",
+            "ifname": "PortChannel0001",
+            "distance": "10",
+            "nexthop-vrf": "nh_vrf",
+            "blackhole": "false",
+        }),
+        True,
+        [
+            "ip route 10.1.1.0/24 10.0.0.57 PortChannel0001 10 nexthop-vrf nh_vrf"
+        ]
+    )
+
+def test_set_blackhole():
+    mgr = constructor()
+    set_del_test(
+        mgr,
+        "SET",
+        ("default|10.1.2.0/24", {
+            "nexthop": "10.0.0.57",
+            "ifname": "PortChannel0001",
+            "distance": "10",
+            "nexthop-vrf": "nh_vrf",
+            "blackhole": "true",
+        }),
+        True,
+        [
+            "ip route 10.1.2.0/24 blackhole 10"
```

Comment:
Does this feature consider mgmt route as well? e.g mgmt route in default & mgmt VRFs.

Reply:
The scope of this PR is to cover STATIC_ROUTE table in config_db. Routes that are reflected in this table are considered, otherwise not.

Reply:
I see all test cases are with PortChannel0001/PortChannel0002, can we have some tests with Etherent and Vlan interfaces as well>

Reply:
Added a test case with Ethernet interfaces. Since bgpcfgd is basically a translator from config_db to the commands. The interface type should not make any difference from the unit test perspective.

Reply:
I understand, it is better to test other interfaces as well.